### PR TITLE
[SOL-2046] Remove unnecessary call to Course Catalog Service

### PR DESCRIPTION
**Pull request by marjev**

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @marjev! It looks like you're a member of a company that does contract work for edX. If you're doing this work as part of a paid contract with edX, you should talk to edX about who will review this pull request. If this work is not part of a paid contract with edX, then you should ensure that there is an OSPR issue to track this work in [JIRA](https://openedx.atlassian.net), so that we don't lose track of your pull request.

To automatically create an OSPR issue for this pull request, just visit this link: https://openedx-webhooks.herokuapp.com/github/process_pr?repo=edx%2Fecommerce&number=921

**Comment by marjev** [comment]
@mjfrey, @vkaracic Here's a PR that proposes a removal of `all_products()` method call that is made on the voucher offers API endpoint.

This is WIP because I'll have to fix the tests and increase the coverage. I tagged you so that you may take a look at the general idea of this fix and propose any additional changes that we can do to further increase the performance of this endpoint.

**Comment by vkaracic** [comment]
LGTM

File: ecommerce/extensions/api/v2/views/vouchers.py [hunk 1]
```diff
@@ -75,12 +76,12 @@ def offers(self, request):
             return Response(status=status.HTTP_400_BAD_REQUEST)
 
         try:
-            offers_data = self.get_offers(products, request, voucher)
+            offers_data = self.get_offers(request, voucher)
```
Comment (mjfrey) [inline ecommerce/extensions/api/v2/views/vouchers.py:75]: This is still the same issue.  The call to get_offers is using the DEFAULT_CATALOG_PAGE_SIZE.  This is a problem.  We want to only get the first page.

**Comment by marjev** [comment]
@mjfrey Can you take a final look before this is merged?

**Comment by mjfrey** [comment]
@marjev do we want to add the removal of the call to the catalog service in all_products and reduce the default page size?

**Comment by marjev** [comment]
@mjfrey can you simply reopen your PR?

**Comment by mjfrey** [comment]
:+1:

**Pull request merged**